File: bindings/nodejs/examples/6-restore.js
/**
 * This example restores a secured backup file.
 */

require('dotenv').config();
const { AccountManager } = require('@iota/wallet');
async function run() {
    try {
        const manager = new AccountManager({
            storagePath: './alice-database',
            clientOptions: {
                nodes: [
                    {
                        url: 'https://firefly.h.chrysalis-devnet.iota.cafe/',
                    },
                ],
                localPow: true,
            },
            secretManager: {
                Stronghold: {},
            },
        });

        // Add the path to the file from example 5-backup.js
        // for example: ./backup/2021-02-12T01-23-11-iota-wallet-backup-wallet.stronghold
        const path = './backup'; //"input your backup file"

        await manager.restoreBackup(path, process.env.SH_PASSWORD);
        const account = await manager.getAccount('Alice');
        console.log('Account:', account.getAlias());
    } catch (error) {
        console.log('Error: ' + error);
    }
    process.exit(0);
}

run();

File: bindings/python/examples/6_restore.py
# Copyright 2020 IOTA Stiftung
# SPDX-License-Identifier: Apache-2.0


import iota_wallet as iw
import os
from dotenv import load_dotenv

# Load the env variables
load_dotenv()

# Get the stronghold password
STRONGHOLD_PASSWORD = os.getenv('STRONGHOLD_PASSWORD')

# This example restores a secured backup file.
account_manager = iw.AccountManager(
    storage_path='./alice-database-restored'
)

# NOTE: In real use cases you need to set the password in a safer way, like getting it from env variables
account_manager.set_stronghold_password(STRONGHOLD_PASSWORD)

#  Add the path to the file from example 5-backup.js
#  for example: ./backup/2021-03-04T15-31-04-iota-wallet-backup-wallet.stronghold
backup_file_path = r'./backup/2021-03-31T14-45-23-iota-wallet-backup-wallet.stronghold'

# NOTE: In real use cases you need to set the password in a safer way, like getting it from env variables
account_manager.import_accounts(backup_file_path, STRONGHOLD_PASSWORD)
account = account_manager.get_account('Alice')
print(f'Account: {account.alias()}')
